src/org/zaproxy/zap/extension/replacer/ExtensionReplacer.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.replacer;
+
+import java.awt.Toolkit;
+import java.awt.event.KeyEvent;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.regex.Pattern;
+
+import javax.swing.KeyStroke;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.parosproxy.paros.network.HttpSender;
+import org.zaproxy.zap.network.HttpSenderListener;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+/**
+ * An add-on which provides an easy way to replace strings in requests and responses.
+ * TODO
+ *      Add API endpoints
+ *      Implement for contexts as well
+ * @author psiinon
+ *
+ */
+public class ExtensionReplacer extends ExtensionAdaptor implements HttpSenderListener {
+
+    public static final String NAME = "ExtensionReplacer";
+
+    // The i18n prefix
+    protected static final String PREFIX = "replacer";
+
+    private OptionsReplacerPanel optionsReplacerPanel;
+    private ReplacerParam params; 
+    private ZapMenuItem replacerMenuItem;
+    private static final Logger LOGGER = Logger.getLogger(ExtensionReplacer.class);
+
+    public ExtensionReplacer() {
+        super(NAME);
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        extensionHook.addOptionsParamSet(getParams());
+        HttpSender.addListener(this);
+
+        if (getView() != null) {
+            extensionHook.getHookView().addOptionPanel(getOptionsReplacerPanel());
+            extensionHook.getHookMenu().addToolsMenuItem(getReplacerMenuItem());
+        }
+
+    }
+
+    @Override
+    public boolean canUnload() {
+        // The extension can be dynamically unloaded, all resources used/added can be freed/removed from core.
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        super.unload();

```

Comment:
I dont think it can until we update the zap jar which would mean we cant release it on 2.5.0
Maybe return false for canUnload() until then?